Tools/peg_generator/pegen/c_generator.py

```diff
@@ -207,19 +207,23 @@ def visit_StringLeaf(self, node: StringLeaf) -> FunctionCall:
             )
 
     def visit_Rhs(self, node: Rhs) -> FunctionCall:
-        if node in self.cache:
-            return self.cache[node]
         if node.can_be_inlined:
-            self.cache[node] = self.generate_call(node.alts[0].items[0])
+            return self.generate_call(node.alts[0].items[0])
+
+        node_str = f"{node}"
```

Comment:
It would be great if we can abstract this away in a function. Maybe

```
def generate_artificial_rule(node, prefix, new_rule_fn):
    node_str = f"{node}"
    key = f"{prefix}_{node_str}"
    if key in self.cache:
             name = self.cache[key]
     else:
             name = new_rule_fn()
             self.cache[key] = name
       return name
```